app/components/at_mention/at_mention.js

```diff
@@ -40,22 +42,27 @@ class AtMention extends React.PureComponent {
     }
 
     goToUserProfile = () => {
-        const {intl, navigator, theme} = this.props;
-        navigator.push({
-            screen: 'UserProfile',
-            title: intl.formatMessage({id: 'mobile.routes.user_profile', defaultMessage: 'Profile'}),
-            animated: true,
-            backButtonTitle: '',
-            passProps: {
-                userId: this.state.id
-            },
-            navigatorStyle: {
-                navBarTextColor: theme.sidebarHeaderTextColor,
-                navBarBackgroundColor: theme.sidebarHeaderBg,
-                navBarButtonColor: theme.sidebarHeaderTextColor,
-                screenBackgroundColor: theme.centerChannelBg
-            }
-        });
+        const {intl, isSearchResult, navigator, onPostPress, theme} = this.props;
+
+        if (!isSearchResult) {
```

Comment:
Instead of passing up both `isSearchResult` and `onPostPress`, this could probably be done with just the `isSearchResult` prop by making the render function
```
return <Text onPress={!this.props.isSearchResult ? this.goToUserProfile : null}>
```